include/glow/Support/CompilationLog.h

```diff
@@ -0,0 +1,60 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef GLOW_SUPPORT_COMPILATIONLOG_H
+#define GLOW_SUPPORT_COMPILATIONLOG_H
+
+#include "llvm/ADT/SmallString.h"
+#include "llvm/Support/CommandLine.h"
+#include <string>
+
+namespace glow {
+
+/// A class that defines the compilation scope.
+///
+/// Example:
+class CompilationScope {
+public:
+  /// A static stack that maintains the current scope stack.
+  static std::vector<std::string> scopeStack_;
+
+private:
+  std::string scopeName_;
+
+public:
+  /// When the constructor is called, current stack is pushed in the scope
+  /// stack.
+  CompilationScope(std::string scopeName);
+
+  /// When the destructor is called, current stack is popped out from the scope
+  /// stack.
+  ~CompilationScope();
+
+  /// A function that returns the current compilation scope stack.
+  llvm::StringRef getCurrentCompilationScopeStack();
+};
+
+/// Dump the compilation log.
+/// \param logContent the content that will be written into the log file.
+void dumpCompilationLog(std::string logContent);
```

Comment:
passing strings by value is not efficient. Can you go through the APIs and make them either StringRef or const std::string&